sdcm/tester.py

```diff
@@ -572,6 +572,8 @@ def get_cluster_aws(self, loader_info, db_info, monitor_info):
         regions = self.params.get('region_name').split()
 
         def get_root_device_name(ami_image_id):
+            if not ami_image_id:
```

Comment:
this is not good enough,
since the code breaks on the `.split()` call before calling this function

Reply:
this might be better
```diff
diff --git i/sdcm/tester.py w/sdcm/tester.py
index 0a28833d..820d3c4d 100644
--- i/sdcm/tester.py
+++ w/sdcm/tester.py
@@ -618,7 +618,7 @@ class ClusterTester(db_stats.TestStatsMixin, unittest.TestCase):  # pylint: disa
             db_info['type'] = self.params.get('instance_type_db')
         if db_info['disk_size'] is None:
             db_info['disk_size'] = self.params.get('aws_root_disk_size_db', default=None)
-        if db_info['device_mappings'] is None:
+        if db_info['device_mappings'] is None and self.params.get('ami_id_db_scylla'):
             if db_info['disk_size']:
                 db_info['device_mappings'] = [{
                     "DeviceName": get_root_device_name(self.params.get('ami_id_db_scylla').split()[0]),
```